weblog/requesthandler.go

```diff
@@ -7,26 +7,29 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/deis/logger/syslogish"
+	"github.com/deis/logger/logs"
+	"github.com/deis/logger/storage"
 	"github.com/gorilla/mux"
 )
 
 type requestHandler struct {
-	syslogishServer *syslogish.Server
+	storageAdapter storage.Adapter
 }
 
-func newRequestHandler(syslogishServer *syslogish.Server) *requestHandler {
-	return &requestHandler{syslogishServer: syslogishServer}
+func newRequestHandler(storageAdapter storage.Adapter) *requestHandler {
```

Comment:
For a simple demonstration of the principle, see:

https://play.golang.org/p/_gQKhf962b (works as expected)
https://play.golang.org/p/NGqXkvxSd3 (has a bug)